In [1]:
import pandas as pd

/Users/mahiradayal/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pandas/compat/__init__.py:120: UserWarning: Could not import the lzma module. Your installed Python is incomplete. Attempting to use lzma compression will result in a RuntimeError.
  warnings.warn(msg)


"credit" shows the percentage of people who have a credit score under 660 by county in 2020 Q1. See creditscore_scrape.ipynb for methodology on how this data (Equifax Subprime Credit Population FRED Economic Data) was scraped and cleaned.

In [2]:
credit = pd.read_csv("/Users/mahiradayal/masters_project/full_credit.csv")

"banks" shows the banks per county (using FIP codes). See https://colab.research.google.com/drive/1AXF7Y1J829PHL6bQHsxSYZM1kk7gUn64?usp=sharing for methodology on how FDIC banks data was cleaned, combined with census income and population data, merged using FIP codes and how the banks per capita figure was reached. Previous notebooks show how this FDIC data was geocoded and tested using the Google Maps API. 

In [3]:
banks = pd.read_csv("/Users/mahiradayal/masters_project/bankscredit.csv")
banks = banks.drop(columns=['B01003_001M', 'B19013_001M', 'state', 'county'])
banks = banks.rename(columns={"Banks": "Number of Banks"})
banks.head()

FIPS  Number of Banks  POPULATION  MEDIAN_HOUSEHOLD_INCOME  \
0  29189                1      996919                    67420   
1  17043                1      929060                    92809   
2  24033                1      908670                    84920   
3  13089                1      749323                    62399   
4  11001                1      692683                    86420   

   BANKS_PER_CAP*100            Name State  
0           0.000100        St Louis    MO  
1           0.000108         Du Page    IL  
2           0.000110  Prince Georges    MD  
3           0.000133         De Kalb    GA  
4           0.000144      Washington    DC

In [4]:
credit.head()

County Name State  FIPS County Code  \
0     AUTAUGA    AL              1001   
1     BALDWIN    AL              1003   
2     BARBOUR    AL              1005   
3        BIBB    AL              1007   
4      BLOUNT    AL              1009   

                                                Link  \
0  https://fred.stlouisfed.org/series/EQFXSUBPRIME00   
1  https://fred.stlouisfed.org/series/EQFXSUBPRIME00   
2  https://fred.stlouisfed.org/series/EQFXSUBPRIME00   
3  https://fred.stlouisfed.org/series/EQFXSUBPRIME00   
4  https://fred.stlouisfed.org/series/EQFXSUBPRIME00   

                                                 url  \
0  https://fred.stlouisfed.org/series/EQFXSUBPRIM...   
1  https://fred.stlouisfed.org/series/EQFXSUBPRIM...   
2  https://fred.stlouisfed.org/series/EQFXSUBPRIM...   
3  https://fred.stlouisfed.org/series/EQFXSUBPRIM...   
4  https://fred.stlouisfed.org/series/EQFXSUBPRIM...   

                                           q1_credit  credit_percent  
0  <span class="series-meta-observation-value">30...        30.03273  
1  <span class="series-meta-observation-value">24...        24.19225  
2  <span class="series-meta-observation-value">41...        41.07486  
3  <span class="series-meta-observation-value">35...        35.78089  
4  <span class="series-meta-observation-value">27...        27.83199

In [5]:
# Looking at whether there's a correlation between number of banks and credit scores

column_1 = credit["credit_percent"]
column_2 = banks["Number of Banks"]
correlation = column_1.corr(column_2)

In [6]:
correlation

-0.02368768215815885

A negative correlation is useful because our credit value is "percentage of people with low credit scores" so a higher value here means a lower credit score. Taking this as a correlation of 0.024. 